src/utils/SparseLineGrid.h

```diff
@@ -50,6 +50,7 @@ class SparseLineGrid : public SparseGrid<ElemT>
 
     /*! \brief Accessor for getting locations from elements. */
     Locator m_locator;
+    grid_coord_t sign(grid_coord_t z);
```

Comment:
My implementation gives +1 output for 0 input which is more useful for this case, but I think your general form is more common.  So maybe I should call it something else when it doesn't return 0 as the sign of 0?